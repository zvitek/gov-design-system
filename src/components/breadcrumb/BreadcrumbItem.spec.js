import { shallowMount } from '@vue/test-utils'
import GovBreadcrumbItem from '@components/breadcrumb/BreadcrumbItem.vue'

let wrapper

describe('GovBreadcrumbItem', () => {
    beforeEach(() => {
        wrapper = shallowMount(GovBreadcrumbItem)
    })

    it('is called', () => {
        expect(wrapper.name()).toBe('GovBreadcrumbItem')
        expect(wrapper.isVueInstance()).toBeTruthy()
    })

    it('render correctly', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('active item must be tag strong', () => {
        wrapper.setProps({
            tag: 'a',
            active: true
        })
        expect(wrapper.vm.computedTag).toBe('strong')
        expect(wrapper.find('strong').classes()).toContain('gov-title--delta')
    })
})
