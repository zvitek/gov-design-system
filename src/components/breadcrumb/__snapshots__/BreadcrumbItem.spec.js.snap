// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GovBreadcrumbItem render correctly 1`] = `<span class="gov-breadcrumbs__item"><a class="gov-link gov-link--standalone"></a></span>`;
